harness/tests/experiment/fixtures/pytorch_onevar_model.py

```diff
@@ -221,13 +227,165 @@ def build_validation_data_loader(self) -> torch.utils.data.DataLoader:
             raise ValueError(f"unknown dataloader_type: {self.hparams['dataloader_type']}")
 
 
-class OneVarTrialWithApexAmp(OneVarTrial):
+class AMPTestDataset(OnesDataset):
+    STAGES = {
+        "one": 1.0,
+        "zero": 0.0,
+        "small": 2e-14,
+        "large": 2e4,
+    }
+
+    def __init__(self, stages: Iterable[str]) -> None:
+        self.stages = stages
+
+    def __len__(self) -> int:
+        return len(self.stages)
+
+    def _get_stage(self, index: int) -> str:
+        stage = self.stages[index]
+        return stage
+
+    def __getitem__(self, index: int) -> Tuple:
+        stage = self._get_stage(index)
+        return self._get_stage_item(stage)
+
+    def _get_stage_item(self, stage) -> Tuple:
+        for _stage, x in self.STAGES.items():
+            if _stage.lower() == stage.lower():
+                y = x
+                return torch.Tensor([float(x)]), torch.Tensor([float(y)])
+        else:
+            raise ValueError(f"Unrecognized {self.__class__.__name__} stage {stage}")
```

Comment:
This is a bit verbose.  I think since this is test code, you can make stronger assumptions about the behavior the calling code (which is all written by us) and you can make the result simpler to read and understand.

So if you wrote:

```python
def __getitem__(self, index: int) -> Tuple:
    n = self.STAGES[self.stages[index]]
    return torch.Tensor([x]), torch.Tensor([x])
```

so no need for `_get_stage()` and `_get_stage_item()` become unnecessary.

Reply:
I totally agree. I want to say that there used to be a better reason for all this extra stuff before I made some other changes, but I dunno. Anyway, this is done!